### PR TITLE
Add integration test for building Swift benchmarks with SwiftPM

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

File: build-swift-benchmarks-with-swiftpm.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Test that we can build the Swift benchmarks using SwiftPM.
+//
+// REQUIRES: have-swift-benchmarks
+//
+// RUN: rm -rf %t.dir
+// RUN: mkdir -p %t.dir/
+// RUN: cp -R %{swift_benchmarks_path} %t.dir/benchmark
```
Comment (gottesmm) [inline build-swift-benchmarks-with-swiftpm.txt]: Wy copy? Why not just use the benchmark path? Other than that if everything works, LGTM.

File: lit.cfg [hunk 2]
```diff
@@ -152,6 +152,15 @@ if swiftpm_srcdir is None:
 if os.path.exists(swiftpm_srcdir):
     config.available_features.add("have-swiftpm")
     config.substitutions.append( ('%{swiftpm_srcdir}', swiftpm_srcdir) )
+
+# Use the default Swift src layout if Swift the benchmark suite path is not
+# provided as a param.
+swift_benchmarks_path = lit_config.params.get("swift-benchmarks-path")
+if swift_benchmarks_path is None:
+    swift_benchmarks_path = os.path.join(srcroot, "..", "swift", "benchmark")
+if os.path.exists(swift_benchmarks_path):
```
Comment (gottesmm) [inline lit.cfg]: Actually, you may want to use abspath here or something like that.

File: build-swift-benchmarks-with-swiftpm.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+// RUN: rm -rf %t.dir
+// RUN: mkdir -p %t.dir/
+//
+// RUN: %{swift} build --package-path %{swift_benchmarks_path} --build-path %t.dir 2>&1 | tee %t.build-log
+//
+// Check the build log.
+//
+// RUN: %{FileCheck} --check-prefix CHECK-BUILD-LOG --input-file %t.build-log %s
+//
+// CHECK-BUILD-LOG: Linking{{.*}} {{.*}}SwiftBench
```
Comment (gottesmm) [inline build-swift-benchmarks-with-swiftpm.txt]: Actually, can you add to this an invocation. Specifically:

```
./swiftbench --num-iters=1 --tags=
```

Basically just run each test with 1 iteration or something like that to make sure everything works.

Comment (gottesmm) [reply build-swift-benchmarks-with-swiftpm.txt]: I am also fine if you just run 1 test. I would use XorLoop. So:

```
./SwiftBench --num-iters=1 XorLoop
```

Just make sure it exits 0.

**Comment by gottesmm** [comment]
Looks beautiful. Run a test and lets get this party started.

**Comment by aciidb0mb3r** [comment]
Running a cross-repo test here: https://github.com/apple/swift/pull/19142#issuecomment-433806295

**Pull request merged**